pkg/scheduler/internal/cache/cache.go

```diff
@@ -520,8 +521,8 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 	// before Update event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods.Has(key):
 		if currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			klog.Errorf("Pod %v updated on a different node than previously added to.", key)
-			klog.Fatalf("Schedulercache is corrupted and can badly affect scheduling decisions")
+			klog.ErrorS(fmt.Errorf("schedulercache is corrupted and can badly affect scheduling decisions"), "Pod updated on a different node than previously added to", "pod", key)
+			os.Exit(1)
```

Comment:
@serathius what is the replacement for `Fatal`?

Reply:
It's `klog.ErrorS()` and `os.Exit(<exit code>)` so the exit code can be customized.

Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#replacing-fatal-calls